src/logs_syslog_lwt.ml

```diff
@@ -57,33 +60,65 @@ let tcp_reporter ?hostname ip ?(port = 514) ?(truncate = 0) ?(framing = `Null) ?
   connect () >>= function
   | Error e -> Lwt.return (Error e)
   | Ok () ->
+    let transmit =
+      if st = Unix.SOCK_DGRAM then
```

Comment:
(sorry, was away from computer for a week, now slowly catching up on the PRs here -- thanks for them!)
I'm curious about the reasoning behind this conditional.  `Lwt_unix.send` calls `send()`, which does in either case return the number of transmitted bytes.  why is a partial write to a unix domain socket not handled here?

Reply:
There is the same thing in the normal unix version too. My (not necessarily correct) understanding is that a partial write would need the same kind of handling as in the UDP reporter (where the return from `sendto` is also ignored). The Unix domain socket is opened in datagram mode by default (with handling for if you happen to be on an either very old or weird system which does syslog in stream mode on `/dev/log`). Without framing, you can't write anything further on a partial write, since the other hand has no idea when the packet ends. So my understanding is that because we're connection-oriented, like TCP, means `send` is required and not `sendto` but, like UDP, there's not a lot we can do about a partial write.

Since the UDP case doesn't handle a partial write, I didn't put one here - perhaps it should be that both should do some error handling 🙂